packages/dcos-integration-test/extra/test_metrics.py

```diff
@@ -134,6 +134,127 @@ def test_metrics_agents_statsd(dcos_api_session):
             get_metrics_prom(dcos_api_session, agent, expected_metrics)
 
 
+@retrying.retry(wait_fixed=2000, stop_max_delay=300 * 1000)
+def check_metrics_prom(dcos_api_session, node, check_func):
+    """Get metrics from prometheus port on node and run check_func function,
+    asserting that it returns True.
+
+    Retries on non-200 status or failed assertions
+    for up to 300 seconds.
+
+    """
+    response = dcos_api_session.session.request(
+        'GET', 'http://{}:61091/metrics'.format(node))
+    assert response.status_code == 200, 'Status code: {}'.format(response.status_code)
+    assert check_func(response) is True
+
+
+def test_task_metrics_metadata(dcos_api_session):
+    # install marathon on marathon
+    install_response = dcos_api_session.cosmos.install_package('marathon')
+    data = install_response.json()
+    dcos_api_session.marathon.wait_for_deployments_complete()
+
+    wait = True
+    mesos_id = node = ''
+    while wait:
+        master = dcos_api_session.masters[0]
+        state_response = dcos_api_session.get('/state', host=master, port=5050)
+        assert state_response.status_code == 200
+        state = state_response.json()
+        for framework in state['frameworks']:
+            if framework['name'] == 'marathon':
+                for task in framework['tasks']:
+                    if task['name'] == 'marathon-user':
+                        mesos_id = task['slave_id']
+                        wait = False
+                        break
+                break
+
+    for agent in state['slaves']:
+        if agent['id'] == mesos_id:
+            node = agent['hostname']
+            break
+
+    def check_task_metrics(response):
+        for line in response.text.splitlines():
+            if '#' in line:
+                continue
+            if 'marathon-user' in line:
+                return ('service_name="marathon"' in line and
+                        'task_name="marathon-user"' in line)
+        return False
+    check_metrics_prom(dcos_api_session, node, check_task_metrics)
```

Comment:
Instead of passing in a callable for `check_metrics_prom()` to execute in order to make assertions about the response, I think it'd be more straightforward to just get the response and then make assertions about it.

```
response = get_metrics_prom(dcos_api_ession, node)
for line in response.text.splitlines():
    # etc.
```

Reply:
I pulled these checks out into a separate method primarily to use the retry decorator since it sometimes takes some time for the metrics/labels to show up

Reply:
Yeah the retry on getting the metrics and asserting their content is necessary. We can still get that while keeping `get_metrics_prom()` focused on retrying the metrics endpoint until it returns 200, without it taking on the responsibility of making assertions about its content.

```python
@retrying.retry(wait_fixed=2000, stop_max_delay=300 * 1000)
def get_metrics_prom(dcos_api_session, node)
    response = dcos_api_session.session.request(
        'GET', 'http://{}:61091/metrics'.format(node))
    assert response.status_code == 200, 'Status code: {}'.format(response.status_code)
    return response


def test_task_metrics_metadata(dcos_api_session)
    # <snip>
    @retrying.retry(...)
    def check_task_metrics():
        response = get_metrics_prom(dcos_api_ession, node)
        for line in response.text.splitlines():
            # etc.
    check_task_metrics()
    # <snip>
```

Reply:
ahh I see, didn't think to add the decorator to the inner func, will fix 👍